railties/lib/rails/generators.rb

```diff
@@ -156,12 +156,25 @@ def self.invoke(namespace, args=ARGV, config={})
         args << "--help" if args.empty? && klass.arguments.any?(&:required?)
         klass.start(args, config)
       else
-        options     = sorted_groups.map(&:last).flatten
-        suggestions = options.sort_by {|suggested| levenshtein_distance(namespace.to_s, suggested) }.first(3)
-        msg =  "Could not find generator '#{namespace}'. "
-        msg << "Maybe you meant #{ suggestions.map {|s| "'#{s}'"}.join(" or ") }\n"
-        msg << "Run `rails generate --help` for more options."
-        puts msg
+        unless sorted_groups.group_by(&:size).min.first > namespace.length
+          options     = sorted_groups.map(&:last).flatten
+          suggestions = options.select do |option|
+            option.split(/[:_]/).any? { |option_segment|
+              namespace.to_s.split(/[:_]/).any? { |name_segment|
+                2 > levenshtein_distance(name_segment, option_segment)
```

Comment:
I don't love the idea of using an edit distance this low. My original use case, wasn't that I knew how to misspell migrations but fat fingered one, key but I actually typed something like "migrates" or "migrate" i.e. I couldn't remember the right command, we're just using spelling as a metric. Setting to such a low value doesn't catch that.